src/log.js

```diff
@@ -1,57 +1,143 @@
 // We only use this here because we want *something* to output if we fail to
 // initialize the backing logging library
-let debug = require('debug')('taskcluster-logging');
 let _ = require('lodash');
 let bunyan = require('bunyan');
 let assume = require('assume');
 let util = require('util');
+let minimatch = require('minimatch');
 
 let allowedLevels = _.keys(bunyan.levelFromName);
 
-function setupLogger(cfg) {
+/**
+ * Parse a string into a mapping of logger names to levels to determine which
+ * level of logging a given logging module should use.  This is done so that we
+ * can optionally have Trace level for what you're working on but then only
+ * have warnings and higher for the rest of the code.  This is similar in
+ * concept to the debug module, but needs higher granularity because we have
+ * more than one log level.  There is currently no globbing support so you will
+ * have to specify all 'name' values in the variable as fully resolved.  The
+ * special value '*' as a name sets the log level for all levels not mentioned.
+ *
+ * Example Variable: "provisioner:trace, api:info,azure-entities:error, *:fatal"
```

Comment:
I could add full ENV compatibility with DEBUG if we are OK with saying that if you don't specify a level that we default to it being INFO...

I want compatibility to be more about making a couple of small conscious changes rather than just flat out duplicating DEBUG.  I'd call it transitional compatibility not full compatibility.

I'd be open to calling the env LOG instead of LOG_LEVEL, but I think LOG is not super descriptive...